svg.js.d.ts

```diff
@@ -82,58 +105,505 @@ declare module "@svgdotjs/svg.js" {
-        putIn(parent: Dom): Dom;
-        remove(): this;
-        removeElement(element: Element): this;
-        replace(element: Dom): Dom;
-        svg(): string;
+    // SVGNumber.js
+    type NumberUnit = [number, string]
+
+    class Number {
+        constructor();
+        constructor(value: Number);
+        constructor(value: string);
+        constructor(value: number, unit?: any);
+        constructor(n: NumberUnit);
+
+        value: number
+        unit: any
+
         toString(): string;
-        words(text: string): this;
-        writeDataToDom(): this;
+        toJSON(): Object;  // same as toString
+        toArray(): NumberUnit
+        valueOf(): number;
+        plus(number: NumberAlias): Number;
+        minus(number: NumberAlias): Number;
+        times(number: NumberAlias): Number;
+        divide(number: NumberAlias): Number;
+        convert(unit: string): Number;
+        to(a: any): Morphable
+    }
+
+    type NumberAlias = Number | number | string;
+
+    // PathArray.js
+
+    type LineCommand = ['M' | 'm' | 'L' | 'l', number, number] | ['H' | 'h' | 'V' | 'v', number] | ['Z' | 'z']
+
+    type CurveCommand =
+        // Bezier Curves
+        ['C' | 'c', number, number, number, number, number, number] |
+        ['S' | 's' | 'Q' | 'q', number, number, number, number] | ['T' | 't', number, number] |
+        // Arcs
+        ['A' | 'a', number, number, number, number, number, number, number]
+
+    type PathCommand = LineCommand | CurveCommand
+
+    type PathArrayAlias = PathArray | PathCommand[] | (string | number)[] | string;
+
+    class PathArray extends Array<PathCommand> {
+        constructor();
+        constructor(d: ArrayAlias<PathCommand> | PathArrayAlias);
+
+        // TODO: maybe typescript does not need to have such as direct access to this variable.
+        // SVG.JS required direct modify the value of a path array for performance
+        value: PathCommand[]
+
+        move(x: number, y: number): this;
+        size(width: number, height: number): this;
+        equalCommands(other: PathArray): boolean
+        morph(pa: PathArray): this
+        at(pos: number): PathArray
+        parse(array: ArrayAlias<PathCommand> | PathArrayAlias): PathCommand[];
+        bbox(): Box;
+        to(a: any): Morphable
+    }
+
+    // Matrix.js
+    interface TransformData {
+        origin?: number[];
+        scaleX?: number;
+        scaleY?: number;
+        shear?: number;
+        rotate?: number;
+        translateX?: number;
+        translateY?: number;
+        originX?: number;
+        originY?: number;
+    }
+
+    interface MatrixLike {
+        a?: number;
+        b?: number;
+        c?: number;
+        d?: number;
+        e?: number;
+        f?: number;
+    }
+
+    interface MatrixExtract extends TransformData, MatrixLike { }
+
+    type FlipType = 'both' | 'x' | 'y' | boolean
+    type ArrayXY = [number, number]
+    type CoordinateXY = ArrayXY | { x: number, y: number }

```

Comment:
You used that type at the Point interface but also had one constructor extra with only `{ x: number, y: number }`. Couldnt you get rid of that extra constructor?

Reply:
good point :)